src/coreclr/jit/lowerxarch.cpp

```diff
@@ -9415,6 +9415,29 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
                             supportsSIMDScalarLoads = (ival <= 0x3F);
                             supportsGeneralLoads    = supportsSIMDScalarLoads;
                         }
+
+                        if (supportsGeneralLoads && childNode->IsVectorZero() && ((op1 == childNode) || (op2 == childNode)))
```

Comment:
This probably isn't the desired fix and regresses some scenarios.

What we basically have here is:
* zmask: bits 0-3
* count_D: bits 4-5
* count_S: bits 6-7

`count_S` indicates which source index is being inserted while `count_D` indicates the destination index that value is being inserted.

So `Vector128<float> Insert(Vector128<float> value, Vector128<float> data, [ConstantExpected] byte index)` is effectively doing `result = value.WithElement(count_D, data.GetElement(count_S)`

`zmask` is then indicating that individual additional elements are zeroed in addition to this extract and insert.

-------------

I think then that we're missing a bit of logic in https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L2107-L2312

When both `op1` and `op2` are zero, we replace ourselves with constant zero already.

When `op1` is zero, we're correctly setting `zmask` to clear everything except the bit we're inserting: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L2181-L2191, we then mark `op1` as contained (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L11350-L11370), and then in codegen handle this by reusing the target register (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsiccodegenxarch.cpp#L1101-L1106). This all looks correct except that we should be checking `op1->isContained()` rather than `op1Reg == REG_NA`.

When `op2` is zero, we are equally setting `zmask` to handle the zeroing itself (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L2192-L2202), we mark `op2` as contained (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L11371-L11392), and then codegen special cases this by reusing `op1Reg`: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsiccodegenxarch.cpp#L1108-L1115

What's firing is the assert here: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsiccodegenxarch.cpp#L1093-L1097

This is failing because we think that general loads are only containable when `ival <= 0x3F` (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L9412-L9446) and that would be true if this were coming from memory. It's not, however, and so is failing to account for the fact that `op2` is being specially handled, so the assert itself is what's "bad".

The "proper" fix here would likely be to update the assert here to special-case the fact that `zero` is containable for `insertps` and that it isn't coming from memory: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/hwintrinsiccodegenxarch.cpp#L1093-L1097. This could be handled by simply moving those 5 lines after the initial `if (ins == INS_insertps) { }` block